x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java

```diff
@@ -176,7 +181,8 @@
 
     DataType(SQLType sqlType, int size, int defaultPrecision, int displaySize, boolean isInteger,
             boolean isRational, boolean defaultDocValues) {
-        this.esType = name().toLowerCase(Locale.ROOT);
+        this.typeName = name().toLowerCase(Locale.ROOT);
+        this.esType = name().equals("DATETIME") ? "date" : typeName;
```

Comment:
I think a better way would be to ask this parameter for the constructor and add it for each declaration. 
So `DATETIME` would be `"date"`, `INTERVAL_`... `null` while `NULL` would be `"null"`. Potentially add an overloaded constructor to specify esType name which by default would be the enum name lowercase but with the possibility of being overwritten.